src/sources/apache_metrics/parser.rs

```diff
@@ -424,6 +376,17 @@ fn line_to_metrics(
     })
 }
 
+fn parse_numeric_value<T: std::str::FromStr>(key: &str, value: &str) -> Result<T, ParseError>
+where
+    <T as std::str::FromStr>::Err: std::error::Error,
+    <T as std::str::FromStr>::Err: 'static,
```

Comment:
These can combine, and I'm not sure the `as FromStr` is necessary:
```suggestion
    T::Err: std::error::Error + 'static,
```